comms/EclMeta.h

```diff
@@ -0,0 +1,221 @@
+#pragma once
+
+#include <global.h>
+#include <UtilFilesystem.h>
+#include <StringSet.h>
+#include "AttributeType.h"
+#include "RecursiveMutex.h"
+#include "SAXHandlersBase.h"
+#include "DiskAttribute.h"
+
+typedef std::pair<std::_tstring, bool> StringBoolPair;	//  Path, include in -I options
+typedef std::vector<StringBoolPair> WPathVector;
+
+class CEclMetaData : public clib::CLockableUnknown
+{
+protected:
+    boost::filesystem::wpath m_path;
+    std::_tstring m_moduleLabel;
+public:
+    BEGIN_CUNKNOWN
+    END_CUNKNOWN(CUnknown)
+
+    const boost::filesystem::wpath & GetPath() { return m_path; }
+    bool HasPath() { return m_path.size() > 0; }
+
+    std::wstring GetValue(Element e, const std::_tstring key)
+    {
+        std::wstring value = _T("");
+        Element::StringStringMap::const_iterator found = e.m_attr.find(key);
+        if (found != e.m_attr.end())
+            value = found->second;
+        return value;
+    }
+    int GetIntValue(Element e, const std::_tstring key)
+    {
+        int value = 0;
+        Element::StringStringMap::const_iterator found = e.m_attr.find(key);
+        if (found != e.m_attr.end())
+            value = std::stoi(found->second);
+        return value;
+    }
+};
+
+class CEclFolder : public CEclMetaData
+{
+public:
+    CEclFolder(const boost::filesystem::wpath path)
```

Comment:
Should be const ref:  `const boost::filesystem::wpath &path`

Reply:
Done